plugins/rbac/src/components/DeleteRoleDialog.tsx

```diff
@@ -65,12 +67,32 @@ const DeleteRoleDialog = ({
 
   const deleteRole = async () => {
     try {
+      const policies = await rbacApi.getAssociatedPolicies(roleName);
+      let cleanupPoliciesResponse;
+      if (Array.isArray(policies)) {
+        const allowedPolicies = policies.filter(
+          (policy: RoleBasedPolicy) => policy.effect !== 'deny',
+        );
+        if (allowedPolicies.length > 0) {
+          cleanupPoliciesResponse = await rbacApi.deletePolicies(
+            roleName,
+            allowedPolicies,
+          );
+        }
+        if (typeof cleanupPoliciesResponse === 'object') {
```

Comment:
it's good to check this on status over object

`jsonResponse.status !== 200 && jsonResponse.status !== 204`